inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -550,6 +550,23 @@ void GNAGraphCompiler::ConcatPrimitive(InferenceEngine::CNNLayerPtr layer) {
     auto& concatLayerInfo = concat_connection.find(concatLayer->name)->second;
     for (auto &&outLayer : concatLayer->outData.front()->getInputTo()) {
         if ( LayerInfo(outLayer.second).isConcat() ) {
+            auto concatCadidate = outLayer.second;
+            if (LayerInfo(concatCadidate).isNonFunctional()) {
+                // searching for next concat
+                auto isNonFunctional = [](CNNLayerPtr l) {
+                    return LayerInfo(l).isNonFunctional();
+                };
+                if (!CNNNetHasNextLayerSkipCertain(concatCadidate, 0, 0, isNonFunctional)) {
+                    continue;
+                }
+                concatCadidate = CNNNetGetNextLayerSkipCertain(concatCadidate, 0, 0, isNonFunctional).first;
```

Comment:
what if there will be non linear subgraph of reshapes? What functional layer will be ended up with?
Create a case concat->reshape->(out:0)affine, (out:1 )concat